app/renderer/js/utils/dnd-util.js

```diff
@@ -2,30 +2,168 @@
+		time = 8;
+	} else if (opt.id === 'dnd_3') {
+		time = 12;
+	} else {
+		time = -1;
+	}
+	if (time > 0) {
+		dndOffTime.hr += time;
+		if (dndOffTime.hr >= 24) {
+			dndOffTime.hr -= 24;
+			dndOffTime.carry += 1;
+		}
+		ConfigUtil.setConfigItem('dndSwitchOff', dndOffTime);
+	} else {
+		ConfigUtil.setConfigItem('dndSwitchOff', null);
+	}
+	const doneButton = document.createElement('button');
+	doneButton.className = 'dndButton';
+	doneButton.innerHTML = 'Done';
+	doneButton.onclick = () => {
+		setTimeout(() => {
+			timeDisableDNDInput.removeChild(doneButton);
+			actionContainer.removeChild(timeDisableDNDInput);
+			showDNDTimeLeft();
+			checkDNDstate();
+		}, 500);
+	};
+	timeDisableDNDInput.appendChild(doneButton);
+	const cancelButton = document.createElement('button');
+	cancelButton.className = 'dndCancel';
+	cancelButton.innerHTML = 'Cancel';
+	cancelButton.onclick = () => {
+		checkDNDstate(true);
+		timeDisableDNDInput.removeChild(doneButton);
+		actionContainer.removeChild(timeDisableDNDInput);
+	};
+	timeDisableDNDInput.appendChild(cancelButton);
+}
+
+function sleep(ms) {
+	return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+function formatTime(str) {
+	return str.length === 1 ? '0' + str : str;
+}
+
+function checkDNDstate(cancel = false) {
+	const check = ConfigUtil.getConfigItem('dndSwitchOff');
+	if (check !== null && typeof check === 'object' && !cancel) {
+		if (check.carry && (new Date().getDay() !== check.day)) { // for dnd operations ending after midnight
+			check.carry = 0;
+		}
+		const savedValue = (check.hr + (24 * check.carry)).toString() + formatTime(check.min.toString());
+		const presentValue = (new Date().getHours()).toString() + formatTime((new Date().getMinutes()).toString());
+		if (parseInt(savedValue, 10) < parseInt(presentValue, 10)) {
+			toggle();
+			sleep(1000).then(() => {
+				ipcRenderer.send('forward-message', 'toggle-dnd', dnd, newSettings);
+				ConfigUtil.setConfigItem('dndSwitchOff', null);
+			});
+		}
+		setTimeout(checkDNDstate, 60 * 1000); // keeps running unless DND is switched off by the timer
+	} else if (cancel) {
+		toggle();
+		sleep(500).then(() => {
+			ipcRenderer.send('forward-message', 'toggle-dnd', dnd, newSettings);
+			ConfigUtil.setConfigItem('dndSwitchOff', null);
+		});
+	}
+}
+
+function showDNDTimeLeft() {
+	const check = ConfigUtil.getConfigItem('dndSwitchOff');
+	if (check !== null && typeof check === 'object') {
+		const timeLeft = document.createElement('span');
+		timeLeft.id = 'timeLeft';
+		timeLeft.innerHTML = 'DND off at<br/> <b>' + check.hr + ' hrs ' + check.min + ' mins</b>';
+		timeLeft.className = 'timeLeft';
+		document.getElementById('actions-container').prepend(timeLeft);
+	}
+}
+
 function toggle() {
 	const dnd = !ConfigUtil.getConfigItem('dnd', false);
 	const dndSettingList = ['showNotification', 'silent'];
 	if (process.platform === 'win32') {
 		dndSettingList.push('flashTaskbarOnMessage');
 	}
-
-	let newSettings;
```

Comment:
Any reason to make this a global variable?

Reply:
now changed.